pallets/ajuna-affiliate/src/traits.rs

```diff
@@ -0,0 +1,18 @@
+use sp_runtime::DispatchError;
+
+pub type PalletIndex = u32;
+pub type CallIndex = u8;
+
+pub type ExtrinsicId = (PalletIndex, CallIndex);
+
+pub trait AffiliateHandler<AccountId> {
+	fn add_account_to_affiliatables(account: AccountId);
```

Comment:
Nice, I like this trait. I make the pallet easily reusable in the treasury handler. However, I don't understand why  `add_account_to_affiliatables` is in this trait, should the trait provide read access only?

Reply:
This is because we want other pallets (aka games) to have the final say in to which accounts should be able to use the affiliate system, this is done so that we can tie the system to some gameplay mechanic in those other pallets make use of the affiliate system.

Reply:
Ok, I see this makes sense. However, I would argue that we might want to separate this into two traits then because the rewards' distribution process, in our case tx fee handler, only needs read access.